core/internal/filetransfer/task_default.go

```diff
@@ -0,0 +1,96 @@
+package filetransfer
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+// DefaultTask is the default task to upload/download files
+type DefaultTask struct {
+	OnComplete TaskCompletionCallback
+
+	// FileKind is the category of file being uploaded or downloaded
+	FileKind RunFileKind
+
+	// Path is the local path to the file
+	Path string
+
+	// Name is the name of the file
+	Name string
+
+	// Url is the endpoint to upload to/download from
+	Url string
+
+	// Headers to send on the upload
+	Headers []string
+
+	// Size is the number of bytes to upload
+	//
+	// If this is zero, then all bytes starting at `Offset` are uploaded; if non-zero,
+	// then that many bytes starting from `Offset` are uploaded.
+	Size int64
+
+	// Offset is the beginning of the file segment to upload
+	Offset int64
+
+	// Response is the http.Response from a successful upload or download request.
+	//
+	// This is nil for failed requests, or requests that have not completed.
+	Response *http.Response
+
+	// Error, if any.
+	Err error
+
+	// ProgressCallback is a callback to execute on progress updates
+	ProgressCallback func(int, int)
+
+	// This can be used to cancel the file upload or download if it is no longer needed.
+	Context context.Context
+}
+
+type DefaultUploadTask DefaultTask
+
+func (t *DefaultUploadTask) Execute(fts *FileTransfers) error {
+	return fts.Default.Upload(t)
+}
+func (t *DefaultUploadTask) Complete(fts FileTransferStats) {
+	t.OnComplete.Complete()
+	if fts != nil {
+		fts.UpdateUploadStats(FileUploadInfo{
+			FileKind:      t.FileKind,
+			Path:          t.Path,
+			UploadedBytes: t.Size,
+			TotalBytes:    t.Size,
+		})
+	}
+}
+func (t *DefaultUploadTask) String() string {
+	return fmt.Sprintf(
+		"DefaultUploadTask{FileKind: %d, Path: %s, Name: %s, Url: %s, Size: %d}",
+		t.FileKind, t.Path, t.Name, t.Url, t.Size,
+	)
+}
+func (t *DefaultUploadTask) CaptureError(err error) error {
+	t.Err = err
+	return fmt.Errorf("filetransfer: upload: error uploading to %s: %v", t.Url, err)
```

Comment:
why is it `filetransfer`

Reply:
fair point- I made this the prefix just because that was the previous prefix in the default file transfer, which is the only place that an error really comes from for this task. but it would make more sense to make the prefix `default upload task: ...` and then let the file transfer itself keep whatever prefix it wants

Reply:
or if this an error that you return, don't add any prefix and add the prefix in the relevant location

Reply:
I just moved the prefix back to the file transfer method it originated from to avoid needing to format here, and added a `SetError` function instead